report/methods/method_a.py

```diff
@@ -0,0 +1,199 @@
+        measurement_lines (Dict[int, LineString]): lines which should be considered in the analysis
+        velocity_calculator (VelocityCalculator): velocity calculator
+
+    Returns:
+        Dictionary containing the result of method A for each given measurement line (id of the
+        measurement line is the key)
+    """
+    results = {}
+    for measurement_line_id, configuration in configurations.items():
+        results[measurement_line_id] = _run_method_a(
+            configuration,
+            trajectory,
+            measurement_lines[measurement_line_id],
+            velocity_calculator,
+        )
+    return results
+
+
+def _run_method_a(
+    configuration: ConfigurationMethodA,
+    trajectory: TrajectoryData,
+    measurement_line: LineString,
+    velocity_calculator: VelocityCalculator,
+) -> ResultMethodA:
+    """Run method A for a single measurement line
+
+    Arguments:
+        configuration (ConfigurationMethodA): configuration which should be used for the analysis
+        trajectory (TrajectoryData): trajectory to analyse
+        measurement_line (LineString): line which should be considered in the analysis
+        velocity_calculator (VelocityCalculator): velocity calculator
+
+    Returns:
+        Result of method A (see ResultMethodA for more detailed information)
+    """
+    passing_frames = _compute_passing_frames(trajectory, measurement_line)
+    df_nt = _compute_n_t(trajectory, passing_frames)
+    df_flow = _compute_flow(trajectory, configuration, passing_frames, velocity_calculator)
+    return ResultMethodA(df_nt, df_flow)
+
+
+def _compute_passing_frames(
+    trajectory: TrajectoryData, measurement_line: LineString
+) -> Dict[int, List[int]]:
+    """Compute the frames and who passed the measurement_line
+
+    Arguments:
+        trajectory (TrajectoryData): trajectory to analyse
+        measurement_line (LineString): line which should be considered in the analysis
+
+    Returns:
+        Dict at which frame pedestrians passed the line, with frame as key, and list of pedestrian
+        ids as value
+    """
+    passing_frame = {}
+    prev_positions = trajectory.get_positions(trajectory.get_min_frame())
+    for frame in range(trajectory.get_min_frame() + 1, trajectory.get_max_frame() + 1):
+        current_positions = trajectory.get_positions(frame)
+        passing_agents = []
+        for agent_id, current_position in current_positions.items():
+            if agent_id in prev_positions.keys():
+                step = LineString([prev_positions[agent_id], current_position])
+                if step.intersection(measurement_line) and not measurement_line.contains(
+                    prev_positions[agent_id]
+                ):
+                    passing_agents.append(agent_id)
+        if passing_agents:
+            passing_frame[frame] = passing_agents
+        prev_positions = current_positions
+    return passing_frame
```

Comment:
Input after discussion with Mohcine for performance improvements:
- Try to filter the ped/frames to only make expensive operations when needed
- Binary search for finding relevant frames